bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -236,15 +262,31 @@ private void updateThings() {
         Map<String, String> zoneIdToTcsIdMap = new HashMap<>();
         Map<String, ThingStatus> idToTcsThingsStatusMap = new HashMap<>();
 
-        // First, create a lookup table
-        for (LocationStatus location : apiClient.getInstallationStatus()) {
-            for (GatewayStatus gateway : location.getGateways()) {
-                for (TemperatureControlSystemStatus tcs : gateway.getTemperatureControlSystems()) {
-                    idToTcsMap.put(tcs.getSystemId(), tcs);
-                    tcsIdToGatewayMap.put(tcs.getSystemId(), gateway);
-                    for (ZoneStatus zone : tcs.getZones()) {
-                        idToZoneMap.put(zone.getZoneId(), zone);
-                        zoneIdToTcsIdMap.put(zone.getZoneId(), tcs.getSystemId());
+        EvohomeApiClient localApiClient = apiClient;
+        if (localApiClient != null) {
```

Comment:
Again nesting. :)
```suggestion
        if (localApiClient == null) {
            return;
        }
```

Reply:
This was intentionally, so that the second block at line 297 of this method would still be executed. I changed it as you suggested. 
The consequence is that when the apiclient == null, all thinghandlers do not update their things (reset them to no zones etc). That maybe a good, on the other hand it looks a functionall change to me.

Reply:
Sorry, I missed that, I'll have another look.

Reply:
It doesn't seem to be a functional change since all maps would be empty, thus the called methods would only be provided with null arguments, thus wouldn't do anything. Or am I still missing something?

Reply:
Again, you right. I thought empty HashMap would be passed, but i double checked and null values are passed. So i changed it should be good now.

Reply:
I read read handler code but would need to double-check, but actually I found one update which accepts null arguments and use that to mark thing as offline? If I found the correct handler, actually you are correct that it's a functional change, and probably undesired. But it's very convoluted...

Reply:
fixed